### PR TITLE
Github Merge API - Merge branch

### PR DESCRIPTION
**Pull request by udiudi**
Hi,
I'm using this package for internal tool and needed an ability to create a merge, so I've added it quickly based on [this api doc](https://developer.github.com/v3/repos/merging/#perform-a-merge).

Maybe it can be useful to anyone else.

Cheers!

### CODE REVIEW

File: src/octonode/merge.coffee [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+class Merge extends Base
+
+  constructor: (@repo, @client) ->
+
+  # Create a merge in the repository
+  # '/repos/pksunkara/hub/merges' POST
+  create: (obj, cb) ->
+    @client.post "/repos/#{@repo}/merges", obj, (err, s, b, h) ->
+      return cb(err) if err
+      if s isnt 201 then cb(new Error("Merge info error")) else cb null, b, h
```
Comment (pksunkara) [inline src/octonode/merge.coffee]: `Merge create error`

**Comment by udiudi** [comment]
@pksunkara kk, will do.

File: src/octonode.coffee [hunk 2]
```diff
@@ -24,6 +24,9 @@ octonode = module.exports =
   # [Pr](octonode/pr.html) class for github
   pr: require './octonode/pr'
 
+  # [Merge](octonode/merge.html) class for github
+  merge: require './octonode/merge'
+
```
Comment (pksunkara) [inline src/octonode.coffee]: You don't need this now.

**Comment by udiudi** [comment]
@pksunkara done!
Thank you for you patience!

**Pull request merged**